Loop/Managers/OnboardingManager.swift

```diff
@@ -0,0 +1,383 @@
+            completedOnboardingIdentifiers.append(activeOnboarding.onboardingIdentifier)
+            self.activeOnboarding = nil
+        }
+        resumeOnboarding()
+    }
+
+    private func ensureAuthorization(_ completion: @escaping () -> Void) {
+        ensureNotificationAuthorization {
+            self.ensureHealthStoreAuthorization {
+                self.ensureBluetoothAuthorization(completion)
+            }
+        }
+    }
+
+    private func ensureNotificationAuthorization(_ completion: @escaping () -> Void) {
+        getNotificationAuthorization { authorization in
+            guard authorization == .notDetermined else {
+                completion()
+                return
+            }
+            self.authorizeNotification { _ in completion() }
+        }
+    }
+
+    private func ensureHealthStoreAuthorization(_ completion: @escaping () -> Void) {
+        getHealthStoreAuthorization { authorization in
+            guard authorization == .notDetermined else {
+                completion()
+                return
+            }
+            self.authorizeHealthStore { _ in completion() }
+        }
+    }
+
+    private func ensureBluetoothAuthorization(_ completion: @escaping () -> Void) {
+        guard bluetoothAuthorization == .notDetermined else {
+            completion()
+            return
+        }
+        authorizeBluetooth { _ in completion() }
+    }
+
+    private func complete() {
+        if let completion = completion {
+            self.completion = nil
+            completion()
+        }
+    }
+
+    // MARK: - State
+
+    private func onboardingFromRawValue(_ rawValue: OnboardingUI.RawValue) -> OnboardingUI? {
+        guard let onboardingType = onboardingTypeFromRawValue(rawValue),
+              let rawState = rawValue["state"] as? OnboardingUI.RawState
+        else {
+            return nil
+        }
+
+        return onboardingType.init(rawState: rawState)
+    }
+
+    private func onboardingTypeFromRawValue(_ rawValue: OnboardingUI.RawValue) -> OnboardingUI.Type? {
+        guard let identifier = rawValue["onboardingIdentifier"] as? String else {
+            return nil
+        }
+
+        return onboardingTypeByIdentifier(identifier)
+    }
+
+    private func onboardingTypeByIdentifier(_ identifier: String) -> OnboardingUI.Type? {
+        return pluginManager.getOnboardingTypeByIdentifier(identifier)
+    }
+}
+
+// MARK: - OnboardingDelegate
+
+extension OnboardingManager: OnboardingDelegate {
+    func onboardingDidUpdateState(_ onboarding: OnboardingUI) {
+        precondition(onboarding === activeOnboarding)
+        userDefaults.onboardingManagerActiveOnboardingRawValue = onboarding.rawValue
+    }
+
+    func onboarding(_ onboarding: OnboardingUI, hasNewTherapySettings therapySettings: TherapySettings) {
+        precondition(onboarding === activeOnboarding)
+        loopDataManager.settings.glucoseTargetRangeSchedule = therapySettings.glucoseTargetRangeSchedule
```

Comment:
Consider adding a function to `LoopDataManager` to update `TherapySettings`, for example `updateTherapySettings(_ therapySettings: TherapySettings)`. Then move this code there, and just call:

```
    loopDataManager.updateTherapySettings(therapySettings)
```
there.

My reasoning is: If someone adds another field to TherapySettings, I'm afraid someone will forget to add a line here to copy it over, and it will be a tough bug to find.